### PR TITLE
Arc 78, extends 26 URI scheme Key Reg

### PR DESCRIPTION
**Pull request by tasosbit**
A backwards compatible extension to ARC-26 which specifies the encoding parameters of key registration transactions

### CODE REVIEW

File: ARCs/arc-0078.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+Elements of the query component may contain characters outside the valid range. These are encoded differently depending on their expected character set. The text components (note, xnote) must first be encoded according to UTF-8, and then each octet of the corresponding UTF-8 sequence must be percent-encoded as described in RFC 3986. The binary components (votekey, selkey, etc.) must be encoded with base64url as specified in <a href="https://www.rfc-editor.org/rfc/rfc4648.html#section-5">RFC 4648 section 5</a>.
+
+### ABNF Grammar
+
+```
+algorandurn     = "algorand://" algorandaddress [ "?" keyregparams ]
+algorandaddress = *base32
+keyregparams    = keyregparam [ "&" keyregparams ]
+keyregparam     = [ typeparam / votekeyparam / selkeyparam / sprfkeyparam / votefstparam / votelstparam / votekdparam / noteparam / feeparam / otherparam ]
+typeparam       = "type=keyref"
```
Comment (ori-shem-tov) [inline ARCs/arc-0078.md:33]:
```suggestion
typeparam       = "type=keyreg"
```

File: ARCs/arc-0078.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    "type": "keyreg",
+    "votefst": 1300,
+    "votekd": 100,
+    "votekey:b64": "UU8zLMrFVfZPnzbnL6ThAArXFsznV3TvFVAun2ONcEI=",
+    "votelst": 11300
+  }
+}
+```
+
+Will result in this ARC-26-K encoded URI:
```
Comment (cusma) [inline ARCs/arc-0078.md:148]:
```suggestion
Will result in this ARC-78 encoded URI:
```

File: ARCs/arc-0078.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
```
Comment (cusma) [inline ARCs/arc-0078.md:105]:
```suggestion
Will result in this ARC-78 encoded URI:
```

File: ARCs/arc-0078.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+- note: As in [ARC-26](./arc-0026.md). A URL-encoded default notes field value that the the user interface may optionally make editable by the user.
+
+- fee: Optional. A static fee to set for the transaction in microAlgos. Useful to signal intent to receive participation incentives (e.g. with a 2,000,000 microAlgo transaction fee.)
+
+- (others): optional, for future extensions
+
+### Appendix
+
+This section contains encoding examples. The raw transaction object is presented along with the resulting ARC-26-K URI encoding.
```
Comment (cusma) [inline ARCs/arc-0078.md:80]:
```suggestion
This section contains encoding examples. The raw transaction object is presented along with the resulting ARC-78 URI encoding.
```

**Comment by tasosbit** [comment]
Heads up: Will be adding an example for a keyreg offline txn

**Comment by tasosbit** [comment]
Also to be discussed: should this support the non participation flag/type? I consider this to be irrelevant nowadays, so I would think not

```
{
  "txn": {
    "fee": 1000,
    "fv": 1776273,
    "gh:b64": "kUt08LxeVAAGHnh4JoAoAMM9ql/hBwSoiFtlnKNeOxA=",
    "lv": 1777273,
    "nonpart": true,
    "snd:b64": "D9kc63bT7v015uy5WSluLKtxISSMORopXdqhsLEF+P0=",
    "type": "keyreg"
  }
}
```

**Pull request merged**